File: app/src/main/java/mydebts/android/app/feature/events/ViewBinder.java
package mydebts.android.app.feature.events;

import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import mydebts.android.app.R;

class ViewBinder {
    static void bind(EventsFragment fragment) {
        View rootView = fragment.getView();

        if (rootView == null) {
            return;
        }

        rootView.findViewById(R.id.button_add_event)
                .setOnClickListener(v -> fragment.onAddEventClick());

        fragment.adapter = new EventsAdapter();
        fragment.adapter.setOnEventClickListener(fragment::onEventClick);

        fragment.eventsRecyclerView = (RecyclerView) rootView.findViewById(R.id.list_events);
        fragment.eventsRecyclerView.setLayoutManager(new LinearLayoutManager(rootView.getContext()));
        fragment.eventsRecyclerView.setAdapter(fragment.adapter);

        fragment.emptyView = rootView.findViewById(R.id.text_no_events);
    }
}

File: app/src/main/java/mydebts/android/app/feature/events/ViewBinder.kt
package mydebts.android.app.feature.events

import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

import mydebts.android.app.R
import mydebts.android.app.ui.OnItemClickListener
import mydebts.android.app.ui.OnItemLongClickListener

internal object ViewBinder {
    fun bind(fragment: EventsFragment) {
        val rootView = fragment.view ?: return

        rootView.findViewById(R.id.button_add_event)
                .setOnClickListener { _ -> fragment.onAddEventClick() }

        fragment.adapter.setOnItemClickListener(object: OnItemClickListener {
            override fun onItemClick(position: Int) {
                fragment.onItemClick(position)
            }
        })
        fragment.adapter.setOnItemLongClickListener(object: OnItemLongClickListener {
            override fun onItemLongClick(position: Int): Boolean {
                return fragment.onItemLongClick(position)
            }
        })

        fragment.eventsRecyclerView = rootView.findViewById(R.id.list_events) as RecyclerView
        fragment.eventsRecyclerView?.layoutManager = LinearLayoutManager(rootView.context)
        fragment.eventsRecyclerView?.adapter = fragment.adapter

        fragment.emptyView = rootView.findViewById(R.id.text_no_events)
    }
}
